tests/aws/services/stepfunctions/v2/callback/test_callback.py

```diff
@@ -773,3 +776,78 @@ def _get_message_body():
             definition,
             exec_input,
         )
+
+    @markers.aws.validated
+    def test_sync_with_task_token(
+        self,
+        aws_client,
+        sqs_create_queue,
+        sqs_send_task_success_state_machine,
+        create_iam_role_for_sfn,
+        create_state_machine,
+        sfn_snapshot,
+    ):
+        sfn_snapshot.add_transformer(
+            JsonpathTransformer(
+                jsonpath="$..output.StartDate",
+                replacement="start-date",
+                replace_reference=False,
+            )
+        )
+        sfn_snapshot.add_transformer(
+            JsonpathTransformer(
+                jsonpath="$..output.StopDate",
+                replacement="stop-date",
+                replace_reference=False,
+            )
+        )
+        sfn_snapshot.add_transformer(
+            JsonpathTransformer(
+                jsonpath="$..MessageId",
+                replacement="message_id",
+                replace_reference=True,
+            )
+        )
+        sfn_snapshot.add_transformer(
+            JsonpathTransformer(
+                jsonpath="$..TaskToken",
+                replacement="task_token",
+                replace_reference=True,
+            )
+        )
+        sfn_snapshot.add_transformer(sfn_snapshot.transform.sfn_sqs_integration())
+
+        queue_name = f"queue-{short_uid()}"
+        queue_url = sqs_create_queue(QueueName=queue_name)
+        sfn_snapshot.add_transformer(RegexTransformer(queue_url, "sqs_queue_url"))
+        sfn_snapshot.add_transformer(RegexTransformer(queue_name, "sqs_queue_name"))
+
+        sqs_send_task_success_state_machine(queue_url)
+
+        template_target = BT.load_sfn_template(ST.SQS_SEND_MESSAGE_AND_WAIT)
+        definition_target = json.dumps(template_target)
+        state_machine_arn_target = create(
+            create_iam_role_for_sfn,
+            create_state_machine,
+            sfn_snapshot,
+            definition_target,
+        )
+
+        template = CT.load_sfn_template(CT.SFN_START_EXECUTION_SYNC_WITH_TASK_TOKEN)
+        definition = json.dumps(template)
+
+        exec_input = json.dumps(
+            {
+                "StateMachineArn": state_machine_arn_target,
+                "Name": "TestStartTarget",
+                "QueueUrl": queue_url,
+            }
+        )
+        create_and_record_execution(
```

Comment:
This starts a StepFunction, which then invokes another target StepFunction (i.e., SQS_SEND_MESSAGE_AND_WAIT) using a sync invoke with task token, right?
It might be worth commenting because this behavior is quite implicit and not readily visible from reading over the test case (without diving in to the SFN definitions).